src/stratigraphy/depth/a_to_b_interval_extractor.py

```diff
@@ -0,0 +1,83 @@
+"""Contains logic for finding AToBInterval instances in a text."""
+
+import re
+
+import fitz
+
+from stratigraphy.lines.line import TextLine
+
+from .depthcolumnentry import DepthColumnEntry
+from .interval import AToBInterval
+from .util import value_as_float
+
+
+class AToBIntervalExtractor:
+    """Methods for finding AToBInterval instances (e.g. "0.5m - 1.8m") in a text."""
+
+    @classmethod
+    def from_lines(cls, lines: list[TextLine]) -> AToBInterval | None:
```

Comment:
Example for boreprofile with this interval structure: 12306.pdf in geoquad/validation folder:
first layer with sublayers being :
0.00 - 0.08 m : Revêtement bitumineux
0.08- 0.30 m : Grave d'infrastructure
0.30 - 1.40 m : Grave dans importante matrice de sable
moyen, brun beige, pulvérulent. 
-> AtoBInterval will be 0.00 -1.40m